lib/wor/paginate/paginate.rb

```diff
@@ -18,9 +20,17 @@ def render_paginated(content, options = {})
     end
 
     def paginate(content, options = {})
+      current_url = request.original_url
+
+      if (preserve_records = options[:preserve_records])
+        content, current_url = Wor::Paginate::Utils::PreserveRecordsHelper
```

Comment:
:thinking: I'm not really sure, but I think this helper should be called on ActiveRecord adapter

Reply:
mhhh no